### DISCUSSION TITLE
Add an alert or warning when multiple authors are editing the same entry

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8106] [Ideas] [answered]
> Created by: brad (brad@pixelandtonic.com) on 2015/03/06 19:07:45 +0000
> Votes at time of UserVoice import: 70

-----------------------------------

Currently if multiple authors are editing the same entry the last person to save is going to win and blow away any other author's changes.

It's not so bad if you're on Craft Pro because you can pull the other author's edits from past revisions (assuming they're enabled), but it would be nice to at least be warned if multiple people are editing the same entry.

**Comment by angrybrad** [comment]
> Posted by Matt Stein (matt@workingconcept.com) on 2016/04/13 16:12:29 +0000
-----------------------------------

FWIW Marion wrote a plugin for warning users that might be editing the same entries: https://github.com/marionnewlevant/craft-snitch

**Comment by angrybrad** [comment]
> Posted by Andre Goldstein (andre@milktop.co.uk) on 2016/02/17 20:11:53 +0000
-----------------------------------

This would definitely be a big game changer when having multiple authors

**Comment by angrybrad** [comment]
> Posted by Billy Patel (billy@billymedia.co.uk) on 2016/05/26 10:52:29 +0000
-----------------------------------

Def need this. I&#39;ve already had the &quot;...but wordpress does it...&quot; email. :)

**Comment by angrybrad** [comment]
> Posted by Reinhard (dassel@dasselundwagner.de) on 2016/04/29 14:35:27 +0000
-----------------------------------

Definitely a must feature. An entry beeing worked on must be blocked for any other user. We want to use craft to publish digital magazines with a bunch of people editing. How can we set up a working process if this feature is not supported by the editing environment?

**Comment by angrybrad** [comment]
> Posted by Florian (fw@demodern.de) on 2015/03/13 06:38:29 +0000
-----------------------------------

Yup this is more than important....its basically a killer no-go for using craft in any halfly proffesionall enviroment - where multiple authors are using the system ..... sadly :(

**Comment by groe** [comment]
No more UserVoice? Can I **+1** here then? :slightly_smiling_face:

I've recently had this request from a client coming from the OpenCMS world (which allows users to "check out" pages, which are then locked for other users).

**Comment by angrybrad** [comment]
@groe Yup, no more UserVoice. You can +1/comment here, though and we imported the number of votes the issue had on UserVoice at the time of import in the initial request.

> Votes at time of UserVoice import: 70

**Comment by brandonkelly** [comment]
@groe to +1 this, add a 👍 reaction to the main post at the top of the page.

**Comment by groe** [comment]
@brandonkelly Done, thanks!

(Now that this is an official @brandonkelly -thumbs-upped issue it can be considered basically finished & almost shipped, RIGHT? :smile: )

**Comment by lukakemperle** [comment]
+1 👍

**Comment by ghost** [comment]
This is a pretty big issue for sites with multiple users

**Comment by j-greig** [comment]
+1 for this... actually a client asked about it during a demo and I had to run a test on the spot to find the answer :grimace:

**Comment by jan-dh** [comment]
+1 on this. Having clients asking for this as well.

**Comment by marionnewlevant** [comment]
[Snitch Plugin](https://plugins.craftcms.com/snitch) does exactly this.

**Comment by peteralewis** [comment]
This is still not baked into the core Craft then?!? :(

**Comment by sjelfull** [comment]
I hope this won't use polling but will rather use some kind of "presence" API, which we can use for other things too.

**Comment by brandonkelly** [comment]
PHP and WebSockets aren’t the best of friends, so we’ll likely do something using polling as the lowest-common-denominator implementation, and spruce it up with an optional server component down the road that uses WebSockets.

**Comment by sjelfull** [comment]
Yeah, I know that. A server component that could provide generic events over WS would be awesome, and highly appreciated. 

This ties into my question about: Will you support GraphQL subscriptions at some point, which will support live streaming updates to things like a official Gatsby source plugin? (please). If so, you will need something like a WS server. But I digress, its another thread.

**Comment by brandonkelly** [comment]
Yep. Work on all of this is already underway ;)

**Comment by ryanmasuga** [comment]
Client just ran into this today as 3 people were editing Matrix blocks on the same entry and apparently blowing away one another's changes. Will probably install Snitch for now, but core capability would be nice.

**Comment by davidhellmann** [comment]
We run also in this problem with a complex multisite setup. 
Is this something that is related to: https://github.com/craftcms/cms/discussions/8134 🤔

**Reply by brandonkelly** [reply]
#8134 is requesting the ability for multiple users to each be listed as authors of a single entry.

**Comment by sm9** [comment]
We've just been asked by a client if Craft has a feature that would alert them if another user was editing the same entry. We immediately thought of the [Snitch](https://plugins.craftcms.com/snitch) plugin, but when we checked this, we noticed that this hasn't been ported over to Craft 4 yet, so that is not an option for us at the moment. We're not sure if there's any other options for this, but we couldn't find any. Would be nice if this was a native feature of course - perhaps in Craft 5? 😊

**Reply by brandonkelly** [reply]
Yep, I’m hoping we can get something in for 4.5, even.

**Reply by sm9** [reply]
Thanks Brandon! 😊

**Comment by brandonkelly** [comment, marked as answer]
This is coming in Craft 4.5 (#13420) :tada:

**Reply by brandonkelly** [reply]
Craft 4.5.0 is out now!

**Comment by dominikkrulak** [comment]
If not yet suggested or already implemented, it also would be nice to notify an author that someone else **made** changes to the entry. So the author who's currently editing the entry knows that there's a change. 

That's a bit bad UX because to see changes the author must reload the entry (where she/he loose edited content) or open the entry in new tab or window just to see changes.

I think I'd solve it by inserting a modal button for each updated field where modal would contain an updated content.

**Reply by brandonkelly** [reply]
@dominikkrulak Update notifications were added as part of the same PR :)
